app/src/main/java/de/s3xy/architecturesample/github/GithubInteractor.java

```diff
@@ -0,0 +1,81 @@
+package de.s3xy.architecturesample.github;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import de.s3xy.architecturesample.github.model.RepositoriesSearchResult;
+import de.s3xy.architecturesample.network.NetworkConnectivityManager;
+import de.s3xy.architecturesample.network.NetworkModule;
+import de.s3xy.architecturesample.network.NoNetworkException;
+import rx.Observable;
+
+/**
+ * Created by Vlad Fedorenko <vfedo92@gmail.com>
+ * 08.26.2016
+ */
+public class GithubInteractor {
+    private GithubApi mGithubAuthApi;
```

Comment:
This instances have different base endpoints and different interceptors. But the best way to move all network stuff to repository, so interactor will be responsible only for business rules but not for web calls.
We will refactor it soon :)